component/common/src/main/java/org/wso2/carbon/extension/identity/emailotp/common/util/OneTimePasswordUtils.java

```diff
@@ -0,0 +1,249 @@
+/*
+ * Copyright (c) 2021, WSO2 Inc. (http://www.wso2.com).
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.carbon.extension.identity.emailotp.common.util;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.wso2.carbon.extension.identity.emailotp.common.constant.Constants;
+import org.wso2.carbon.extension.identity.emailotp.common.exception.EmailOtpServerException;
+
+import java.security.InvalidKeyException;
+import java.security.NoSuchAlgorithmException;
+import java.security.SecureRandom;
+
+import javax.crypto.Mac;
+import javax.crypto.spec.SecretKeySpec;
+
+/**
+ * OTP generation utils.
+ * <p>
+ * Important: This class duplicates OTP util methods of the authenticator
+ * module's 'org.wso2.carbon.identity.authenticator.emailotp.OneTimePassword' class.
+ * Any fixes here, should be reflected there as well.
+ */
+public class OneTimePasswordUtils {
+
+    // Each OTP digit(0-9) has a corresponding checksum digit as in the below array.
+    private static final int[] CHECKSUM_DIGITS = {0, 2, 4, 6, 8, 1, 3, 5, 7, 9};
+
+    private static final Log log = LogFactory.getLog(OneTimePasswordUtils.class);
+
+    public static String getRandomNumber(int size) {
```

Comment:
Its better to add a method comment since this is a public method. This is more of a `generateOTP` or `generateToken` right? If that's the case shall we update the method?

Reply:
Also make sure the implementation is compatible with https://wso2.com/technical-reports/wso2-secure-engineering-guidelines/#A53

Reply:
This class is actually a copy of `org.wso2.carbon.identity.authenticator.emailotp.OneTimePassword` , So in case if this is going to use in future as common one, didn't do the changes.